src/org/opendatakit/briefcase/export/ExportToCsv.java

```diff
@@ -0,0 +1,153 @@
+
+    // Get a Map with all the files we need to produce:
+    // - One per repeat group
+    // - One for the main CSV file
+    Map<Model, OutputStreamWriter> files = repeatGroups.stream().collect(toConcurrentMap(
+        group -> group,
+        group -> getOutputStreamWriter(
+            exportDir.resolve(stripIllegalChars(formDef.getFormName()) + "-" + group.getName() + ".csv"),
+            configuration.getOverwriteExistingFiles().orElse(true),
+            getRepeatHeader(group)
+        )
+    ));
+    OutputStreamWriter mainFile = getOutputStreamWriter(
+        exportDir.resolve(stripIllegalChars(formDef.getFormName()) + ".csv"),
+        configuration.getOverwriteExistingFiles().orElse(true),
+        getMainHeader(formDef.getModel(), formDef.isFileEncryptedForm())
+    );
+    files.put(formDef.getModel(), mainFile);
+
+    // Parse all submission files in the instances folder of this form
+    SubmissionParser
+        .parseAllInFormDir(
+            formDef.getFormDir(),
+            formDef.isFileEncryptedForm(),
+            configuration.getPrivateKey(),
+            configuration.getDateRange()
+        )
+        // While we iterate over each submission, take a peek, and
+        // write lines on each repeat group CSV file
+        .peek(submission -> repeatGroups.forEach(groupModel -> append(getRepeatCsvLine(
+            groupModel,
+            submission.getElements(groupModel.fqn()),
+            exportMedia,
+            configuration.getExportMediaPath(),
+            submission.getInstanceId(),
+            submission.getWorkingDir()
+        ), files.get(groupModel))))
+        // Sort the parsed submissions using a long instead of an OffsetDateTime
+        // to get a little boost of performance
+        .sorted(comparingLong(Submission::getSubmissionDateEpoch))
+        // Write lines in the main CSV file
+        .forEachOrdered(submission -> {
+          // Increment the export count and maybe report progress
+          exportTracker.incAndReport();
+          append(getMainSubmissionLines(
+              submission,
+              formDef.getModel(),
+              formDef.isFileEncryptedForm(),
+              exportMedia,
+              configuration.getExportMediaPath()
+          ), mainFile);
+        });
+
+    // Flush and close output streams
+    files.values().forEach(UncheckedFiles::close);
+
+    exportTracker.end();
+
+    ExportOutcome exportOutcome = exportTracker.computeOutcome();
+    if (exportOutcome == ALL_EXPORTED)
+      EventBus.publish(ExportEvent.successForm(formDef, (int) exportTracker.total));
+
+    if (exportOutcome == SOME_SKIPPED)
+      EventBus.publish(ExportEvent.partialSuccessForm(formDef, (int) exportTracker.exported, (int) exportTracker.total));
+
+    if (exportOutcome == ALL_SKIPPED)
+      EventBus.publish(ExportEvent.failure(formDef, "All submissions have been skipped"));
+
+    return exportOutcome;
+  }
```

Comment:
If the above is your personal preference, fine, but you might consider 
```
switch (exportOutcome) {
  case ALL_EXPORTED:
    EventBus.publish(ExportEvent.successForm(formDef, (int) exportTracker.total));
    break;
  case SOME_SKIPPED:
    EventBus.publish(ExportEvent.partialSuccessForm(formDef, (int) exportTracker.exported, (int) exportTracker.total));
    break;
  case ALL_SKIPPED:
    EventBus.publish(ExportEvent.failure(formDef, "All submissions have been skipped"));
    break;
}
```
or even
```
Object event = null;
switch (exportOutcome) {
  case ALL_EXPORTED:
    event = ExportEvent.successForm(formDef, (int) exportTracker.total);
    break;
  case SOME_SKIPPED:
    event = ExportEvent.partialSuccessForm(formDef, (int) exportTracker.exported, (int) exportTracker.total);
    break;
  case ALL_SKIPPED:
    event = ExportEvent.failure(formDef, "All submissions have been skipped");
    break;
}
EventBus.publish(event);
```

Reply:
But my if expression takes also into account the number of processed submissions... How can we express that in a `switch`?

Reply:
The switch on the enum with no default makes it explicit that we are addressing all three mutually exclusive conditions. For what that’s worth.

Reply:
Using else/if’s would get you partway there. 
I don’t care much about this one.